include/deal.II/particles/particle_handler.h

```diff
@@ -718,20 +718,81 @@ namespace Particles
+     * between shared and distributed triangulations. This function should be
+     * used like the corresponding function with the same name in the
+     * SolutionTransfer() class.
+     */
+    void
+    prepare_for_coarsening_and_refinement();
+
+    /**
+     * This function prepares the particle handler for a coarsening and
+     * refinement cycle, by storing the necessary information to transfer
+     * particles to their new cells. The implementation depends on the
+     * triangulation type that is connected to the particle handler and differs
+     * between shared and distributed triangulations. This function should be
+     * used like the SolutionTransfer::interpolate() function after mesh
+     * refinement has finished. Note that this function requires a working
+     * mapping, i.e. if you use a mapping class that requires setup after
+     * a mesh refinement (e.g. MappingQCache(), or MappingEulerian()), the
+     * mapping has to be ready before you can call this function.
+     *
+     * @note It is important to note, that if you do not call this function
+     * after a refinement/coarsening operation (and a repartitioning operation
+     * in a distributed triangulation) and the particle handler contained
+     * particles before, the particle handler will not be usable any more. Not
+     * calling this function after refinement would be similar to trying to
+     * access a solution vector after mesh refinement without first using a
+     * SolutionTransfer class transfer the vector to the new mesh.
      */
     void
+    transfer_after_coarsening_and_refinement();
+
+    /**
+     * This function prepares the particle handler for serialization.
+     * The implementation depends on the triangulation type that is
+     * connected to the particle handler and differs between shared and
+     * distributed triangulations. This function should be used like the
+     * corresponding function with the same name in the SolutionTransfer()
+     * class.
+     *
+     * @note It is important to note, that if you do not call this function
+     * before a serialization operation in a distributed triangulation, the
+     * particles will not be serialized.
+     */
+    void
+    prepare_for_serialization();
+
+    /**
+     * This function prepares the particle handler for serialization.
+     * The implementation depends on the triangulation type that is
+     * connected to the particle handler and differs between shared and
+     * distributed triangulations. This function should be used like the
+     * corresponding function with the same name in the SolutionTransfer()
+     * class.
+     */
+    void
+    deserialize();
+
+    /**
+     * @deprecated: Please use prepare_for_coarsening_and_refinement() or
+     * prepare_for_serialization() instead. See there for further information
+     * about the purpose of this function.
+     */
+    DEAL_II_DEPRECATED
+    void
     register_store_callback_function();
```

Comment:
Could you mark this function `DEAL_II_DEPRECATED_EARLY` according to this [changelog](https://github.com/dealii/dealii/blob/787280118173e892de6a255343638a988a24d806/doc/news/9.2.0-vs-9.3.0.h#L1131-L1144) entry.

Also, feel free to keep the original documentation of this function before the `@deprecated` tag.